src/api/api.c

```diff
@@ -949,6 +966,9 @@ int8_t z_query_reply(const z_query_t *query, const z_keyexpr_t keyexpr, const ui
                             },
                         .encoding = {.prefix = opts.encoding.prefix, .suffix = opts.encoding.suffix}};
     return _z_send_reply(query, keyexpr, value);
+#else
+    return _Z_ERR_GENERIC;
+#endif
 }
 
 _Bool z_reply_is_ok(const z_owned_reply_t *reply) {
```

Comment:
If queryables are disabled, I would completely omit any related APIs.

In doing so, the user will have an explicit error at compilation. Otherwise, it will become harder for users (and us) to debug non-compliant behavior.

Also, you can save some extra bytes in the final binary.